api/include/opentelemetry/baggage/baggage.h

```diff
@@ -0,0 +1,304 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "opentelemetry/common/kv_properties.h"
+#include "opentelemetry/nostd/shared_ptr.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+
+namespace baggage
+{
+
+class Baggage
+{
+public:
+  static constexpr size_t kMaxKeyValuePairs = 180;
+  static constexpr size_t kMaxKeyValueSize  = 4096;
+  static constexpr size_t kMaxSize          = 8192;
+  static constexpr char kKeyValueSeparator  = '=';
+  static constexpr char kMembersSeparator   = ',';
+  static constexpr char kMetadataSeparator  = ';';
+
+  Baggage() : kv_properties_(new opentelemetry::common::KeyValueProperties()) {}
+  Baggage(size_t size) : kv_properties_(new opentelemetry::common::KeyValueProperties(size)){};
+
+  template <class T>
+  Baggage(const T &keys_and_values)
+      : kv_properties_(new opentelemetry::common::KeyValueProperties(keys_and_values))
+  {}
+
+  static nostd::shared_ptr<Baggage> GetDefault()
+  {
+    static nostd::shared_ptr<Baggage> baggage{new Baggage()};
+    return baggage;
+  }
+
+  /* Get value for key in the baggage
+     @returns true if key is found, false otherwise
+  */
+  bool GetValue(const nostd::string_view &key, std::string &value) const
```

Comment:
sorry if I have confused you. But can we remove the const reference to nostd::string_view from all the places as signature, as it doesn't really make sense? Just copying nostd::string_view by value is enough.

Reply:
Done